beacon_node/store/src/database/leveldb_impl.rs

```diff
@@ -0,0 +1,276 @@
+            (
+                BytesKey::from_vec(get_key_for_col(column.as_str(), Hash256::zero().as_bytes())),
+                BytesKey::from_vec(get_key_for_col(
+                    column.as_str(),
+                    Hash256::repeat_byte(0xff).as_bytes(),
+                )),
+            )
+        };
+
+        for (start_key, end_key) in [
+            endpoints(DBColumn::BeaconStateTemporary),
+            endpoints(DBColumn::BeaconState),
+            endpoints(DBColumn::BeaconStateSummary),
+        ] {
+            self.db.compact(&start_key, &end_key);
+        }
+
+        Ok(())
+    }
+
+    pub fn compact_column(&self, column: DBColumn) -> Result<(), Error> {
+        // Use key-size-agnostic keys [] and 0xff..ff with a minimum of 32 bytes to account for
+        // columns that may change size between sub-databases or schema versions.
+        let start_key = BytesKey::from_vec(get_key_for_col(column.as_str(), &[]));
+        let end_key = BytesKey::from_vec(get_key_for_col(
+            column.as_str(),
+            &vec![0xff; std::cmp::max(column.key_size(), 32)],
+        ));
+        self.db.compact(&start_key, &end_key);
+        Ok(())
+    }
+
+    pub fn iter_column_from<K: Key>(
+        &self,
+        column: DBColumn,
+        from: &[u8],
+        predicate: impl Fn(&[u8], &[u8]) -> bool + 'static,
+    ) -> ColumnIter<K> {
+        let start_key = BytesKey::from_vec(get_key_for_col(column.into(), from));
+
+        let iter = self.db.iter(self.read_options());
+        iter.seek(&start_key);
+
+        Ok(Box::new(
+            iter.take_while(move |(key, _)| {
+                metrics::inc_counter_vec(&metrics::DISK_DB_READ_COUNT, &[column.into()]);
+                let Some(trimmed_key) = key.remove_column_variable(column) else {
+                    return false;
+                };
+                let Some(trimmed_start_key) = start_key.remove_column_variable(column) else {
+                    return false;
+                };
+                key.matches_column(column) && predicate(trimmed_key, trimmed_start_key)
+            })
+            .map(move |(bytes_key, value)| {
+                let key = bytes_key.remove_column_variable(column).ok_or_else(|| {
+                    HotColdDBError::IterationError {
+                        unexpected_key: bytes_key.clone(),
+                    }
+                })?;
+                Ok((K::from_bytes(key)?, value))
+            }),
+        ))
+    }
+
+    pub fn iter_column_keys_from<K: Key>(&self, column: DBColumn, from: &[u8]) -> ColumnKeyIter<K> {
+        let start_key = BytesKey::from_vec(get_key_for_col(column.into(), from));
+
+        let iter = self.db.keys_iter(self.read_options());
+        iter.seek(&start_key);
+
+        Ok(Box::new(
+            iter.take_while(move |key| key.matches_column(column))
+                .map(move |bytes_key| {
+                    metrics::inc_counter_vec(&metrics::DISK_DB_READ_COUNT, &[column.into()]);
```

Comment:
I think it's not ideal to count reads of keys (this function) as reads of entries (other uses of `DISK_DB_READ_COUNT`). If this iterators just read keys, maybe we should use a different metric `DISK_DB_KEY_READ_COUNT` or similar

Reply:
yep makes sense, I've added separate metrics for db key read